### PR TITLE
Convert_LinearRobustDemoCombined_to_python

### LINKED ISSUE #203: Convert linregRobustDemoCombined to python
This is figure number 11.9 in the book. its matlab code is [here](https://github.com/probml/pmtk3/blob/master/demos/linregRobustDemoCombined.m)

### PR DESCRIPTION
**Pull request by Drishttii**
@murphyk and @mjsML , please review this code. Here, I have tried to use `sklearn` as much as possible and the `statsmodels` package has been useful for student-t distribution loss. In the statsmodels package, in the files, edgeworth.py, markov_switching.py, kernels.py, functional.py, moment_helpers.py imports like `from scipy.misc import factorial` or from` scipy.misc import logsumexp` should be changed to `scipy.special import factorial/logsumexp` as these functions are located in `scipy.special` (for scipy >=1.3.0)

Closes #203 
Robust

Fig. 11.9 :

### CODE REVIEW

**Comment by murphyk** [comment]
Thanks. Please can you swap out your L1 loss method with [this statsmodel code](https://www.statsmodels.org/dev/examples/notebooks/generated/quantile_regression.html), using quantile=0.5 regression (least absolute deviation). Interestingly,  it seems sklearn is about to add this too - see [this PR](https://github.com/scikit-learn/scikit-learn/pull/9978)

**Comment by Drishttii** [comment]
Yes. Implemented L1 using statsmodels quantregressor, but the predictions made by Huber (delta=1) and quantregressor (q=0.5) are almost similar (difference in the order of `1e-4`), because of the nature of the Huber loss.
